src/Notes.cc

```diff
@@ -343,7 +343,7 @@ AppendTokens(NotePairs::Entries &entries, const SBuf &key, const SBuf &val, cons
     while (tok.token(v, delimiters))
         entries.push_back(new NotePairs::Entry(key, v));
```

Comment:
The posted changes fix bug 5417 in my tests, but I am posting this as a _draft_ PR because I suspect that we need to do more in this value-splitting loop to fix handling of coma-separated annotation values like `a,b,,d`: Such values should also match an empty note ACL value (when that ACL is using comma as a value delimiter), but, I bet, they currently do not.

@eduard-bagdasaryan, could you please check whether more work is indeed needed here and implement any necessary changes?

Reply:
> Do we need such annotations to me matched against an empty note ACL?

Yes, I think so. If there are no negative side effects, let's be consistent in our handling: If a place in the composite value can be interpreted as holding an empty annotation value, let's interpret that place as such.

One could argue that a _terminating_ delimiter ought to be treated differently because an extra trailing delimiter is not uncommon for annotation-accumulating helpers, but I would not make an exception for that use case, especially if doing so complicates our code: If somebody wants to match empty values (a rare use case!), _they_ should pay the price of carefully composing the composite value, without an extra trailing delimiter.

Reply:
Done (20f447c).